tests/Aspire.Hosting.Tests/Health/ResourceHealthCheckServiceTests.cs

```diff
@@ -563,6 +563,100 @@ await app.ResourceNotifications.PublishUpdateAsync(child.Resource, s => s with
         await app.StopAsync().DefaultTimeout();
     }
 
+    [Fact]
+    public async Task ResourceNotHealthyIfResourceReadyEventIsRunning()
+    {
+        var tcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+
+        using var builder = TestDistributedApplicationBuilder.Create(testOutputHelper);
+
+        // Create a healthcheck that waits before returning healthy
+        builder.Services.AddHealthChecks().AddAsyncCheck("healthcheck_a", async () =>
+        {
+            await tcs.Task;
+            return HealthCheckResult.Healthy();
+        });
+
+        var resource = builder.AddResource(new ParentResource("resource"))
+            .WithHealthCheck("healthcheck_a");
+
+        await using var app = await builder.BuildAsync().DefaultTimeout();
+
+        await app.StartAsync().DefaultTimeout();
+
+        await app.ResourceNotifications.PublishUpdateAsync(resource.Resource, s => s with
+        {
+            State = new ResourceStateSnapshot(KnownResourceStates.Starting, null)
+        }).DefaultTimeout();
+
+        var startingEvent = await app.ResourceNotifications.WaitForResourceAsync("resource", e => e.Snapshot.State?.Text == KnownResourceStates.Starting).DefaultTimeout();
+        Assert.Null(startingEvent.Snapshot.HealthStatus);
+
+        await app.ResourceNotifications.PublishUpdateAsync(resource.Resource, s => s with
+        {
+            State = new ResourceStateSnapshot(KnownResourceStates.Running, null)
+        });
+
+        // Resource is unhealthy because ResourceReadyEvent is running.
+        var runningEvent = await app.ResourceNotifications.WaitForResourceAsync("resource", e => e.Snapshot.State?.Text == KnownResourceStates.Running).DefaultTimeout();
+        Assert.Equal(HealthStatus.Unhealthy, runningEvent.Snapshot.HealthStatus);
```

Comment:
Is it also worth validating that:
`Assert.Equal(HealthStatus.Unhealthy, resource.Resource.HealthStatus);
`?

Reply:
nvm doesn't seem like that i s possible anyway.